docker-app/qfieldcloud/core/tests/test_secret.py

```diff
@@ -0,0 +1,183 @@
+
+        with self.assertRaises(UserProjectRoleError):
+            Secret.objects.for_user_and_project(self.u4, self.p1)
+
+    def test_raises_when_user_is_not_organization_member(self):
+        _s1 = self._create_secret(name="s1", project=self.p1)
+
+        with self.assertRaises(UserOrganizationRoleError):
+            Secret.objects.for_user_and_project(self.u5, self.p1)
+
+    def test_secret_not_available_for_random_org_project(self):
+        _s1 = self._create_secret(name="s1", project=self.p1)
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p2)
+
+        self.assertEqual(secrets.count(), 0)
+
+    def test_assigned_secret_overrides_org_project_secret(self):
+        _project_secret = self._create_secret(name="s1", project=self.p1)
+        assigned_secret = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u1
+        )
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p1)
+
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], assigned_secret)
+
+    def test_multiple_secrets_per_user_in_org_project(self):
+        _s1_project = self._create_secret(name="s1", project=self.p1)
+        s1_assigned = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u1
+        )
+        s2_assigned = self._create_secret(
+            name="s2", project=self.p1, assigned_to=self.u1
+        )
+        s3_project = self._create_secret(name="s3", project=self.p1)
+        _s1_another_project = self._create_secret(
+            name="s1", project=self.p2, assigned_to=self.u1
+        )
+        _s1_another_user = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u2
+        )
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p1)
+
+        self.assertEqual(secrets.count(), 3)
+        self.assertEqual(secrets[0], s1_assigned)
+        self.assertEqual(secrets[1], s2_assigned)
+        self.assertEqual(secrets[2], s3_project)
+
+    def test_ind_project_secret_available_for_owner(self):
+        s1 = self._create_secret(name="s1", project=self.p3)
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p3)
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], s1)
+
+    def test_assigned_secret_overrides_ind_project_secret(self):
+        _s1_project = self._create_secret(name="s1", project=self.p3)
+        s1_assigned = self._create_secret(
+            name="s1", project=self.p3, assigned_to=self.u1
+        )
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p3)
+
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], s1_assigned)
+
+    def test_add_secrets_to_projet_and_user_overrides(self):
+        p = Project.objects.create(name="project", owner=self.u1)
+        self._create_secret(name="s1", project=p)
+        user_project_secret = self._create_secret(
```

Comment:
```suggestion
        s1_assigned = self._create_secret(
```

I just observed that in previous tests this would be the name. No strong opinion, just looking for consistency.

To me it is clear that it is a "project secret" based on the name of the test since there is not organization secret to distinguish with. 

Again, feel free to disagree.